### ISSUE TITLE
Alternative SE Clustering Specifications?

### ISSUE THREAD

**Issue by ammalmbe51** [synth-inference/synthdid#121]
Is there a way to change which variable is used for SE clustering? (I.e. main unit of analysis is counties, but want to cluster at state level).

**Comment by davidahirshberg** [comment]
No, not in the package. You’d have to do a state-level bootstrap yourself.

**Comment by ammalmbe51** [comment]
Thank you for the quick response!

*Alice Malmberg*
[she/her/hers]
Ph.D. Student | Department of Political Science, UC Davis
Graduate Student Researcher and Teaching Assistant | UC Center Sacramento
https://www.alicemalmberg.com

**Comment by JediKnight5** [comment]
Hi @davidahirshberg ,

I am experiencing a similar setup as @ammalmbe51 where the treatment is assigned at the state level, but I want to estimate the treatment effect at the county level for reasons of representativeness and precision gain. (In my setup, counties are perfectly nested within states.)

I was thinking about implementing the following clustered blocked bootstrap procedure:

1) Split the states into treatment and control clusters.
2) For every bootstrap iteration, separately sample states from the two clusters. Afterwards, pull all rows that capture the time series for each county from the counties belonging to the union of sampled treated and control states, and estimate SDID on this resampled data.

Do you mind confirming whether this reflects your view?

Thanks in advance!

**Comment by davidahirshberg** [comment]
Hi @JediKnight5. Clustered bootstraps aren't my area of expertise, so I don't want to give you misleading confirmation, but let me give you a heuristic.  When the number of treated units is relatively small and there's good overlap, so we get a nice dense set of synthetic control weights, the dominant source of variation in the SDID ATT estimate is the sample average of treated units.  Given the paper's assumptions, you can more or less check this by comparing the 'effective number of control units'  $N_{0,eff} = 1/\Vert\omega\rVert_2^2$ to the number of treated units $N_1$; this approximation would work if $N_{0,eff} \gg N_1$.  This means that any procedure that'd estimate the variance of the simple sample average of treated unit outcomes should work for SDID in large samples. 

If your assumptions differ, and it sounds like they might, you may want to think about this a little and test coverage using simulated data. If your simulation results don't line up with your expectations, that'd be good to know. In that case, post some new comments to this thread and I'll follow up.

**Comment by JediKnight5** [comment]
Hi @davidahirshberg , thanks for your prompt response!

Your heuristic is very interesting. I computed $N_{0,eff}$ for my application and found it to be of roughly the same magnitude as $N_0$, indicating a relatively dense set of region weights. However, the ratio of $N_{0,eff}$ to $N_1$ is about 2.7 in my application, which clearly does not satisfy the “much larger” condition.

Since my application involves multiple outcomes, I also computed 
$Var(\bar{Y}) = \frac{1}{(\sum^N_{i=1} \sum^T_{t=1} W_{it})^2} \sum^N_{i=1} \sum^T_{t=1} W_{it}  (Y_{it} - \bar{Y})^2$
and found that the relationship between these variances across the different outcomes is similar to that of my estimated standard errors.

In addition, I conducted a coverage study by estimating a placebo effect 400 times, randomly reshuffling the treatment group indicator in my application and recording the respective estimated ATT p-values. The share of p-values exceeding 5% is 0.945 using my clustered blocked bootstrap procedure, compared to 0.877 when applying the package’s blocked bootstrap.

Thanks again for your help, and please let me know if you’d be interested in me sharing any code.
